### PR TITLE
Remove logging from sendSolver

### PR DESCRIPTION
**Pull request by facundominguez**
The logger makes the interface larger and offers no reuse of code.

If the user wants logging, she can surround the calls to sendSolver herself with logging calls.

### CODE REVIEW

**Comment by qaristote** [comment]
Copying my answer from #7.

---

It's true that it originally made sense because it was part of the larger simple-smt library, and it's not particularly useful here, so I'd be in favor of removing it.

One thing that could be a problem though would be sending commands in lazy mode: logging as you proposed could produce something like
```
[send] (set-option :produce-models true)
[send] (set-option :print-sucess true)) # intentional typo
[send] (assert (= 0 0))
[send] (check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
```
instead of
```
[send] (set-option :produce-models true)
(set-option :print-sucess true)) # intentional typo
(assert (= 0 0))
(check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
```
making debugging harder.

**Review by qaristote — APPROVED** [review]
LGTM. I think we can also remove the `sendSolver` function as it is not exposed and is equivalent to `send (backend solver)`, which isn't much longer to write.

File: Z3/smtlib-backends-z3.cabal [hunk 1]
```diff
@@ -28,7 +28,7 @@ library
   ghc-options:      -Wall -Wunused-packages
   exposed-modules:  SMTLIB.Backends.Z3
   build-depends:
-      base             >=4.15 && <4.17.0
+      base             >=4.14 && <4.17.0
```
Comment (qaristote) [inline Z3/smtlib-backends-z3.cabal:31]: Why is this change necessary ? And if it is not, how did you know you could relax the lower bound on the version of `base` ? I used `4.15` because that's what auto-complete gave me but I wonder if there's a way to know up until which version is a piece of code valid.

**Comment by facundominguez** [comment]
I finally got to build this library. Please excuse me for the multiple build errors here.

I'm not using flakes because `nix develop` wants to download like 8 GBs of tools, and wants to install HLS which I don't use. Instead, I'm using `nix-shell -p z3` and the `ghc` on my path which is `8.10.7`. Therefore, I tested building it with an older base and had to relax the bounds :)

> One thing that could be a problem though would be sending commands in lazy mode: logging as you proposed could produce something like

The only difference before and after in the example seems to be that some messages are not prepended with tags `[send]` and `[recv]`. I'm having some difficulty seeing how that makes debugging easier or harder. Also, this behavior would depend on how the user decides to do logging in the surrounding code.

**Comment by qaristote** [comment]
> I'm not using flakes because `nix develop` wants to download like 8 GBs of tools

This might be because you haven't [set up your binary cache for use with `haskell.nix](https://github.com/input-output-hk/haskell.nix/blob/master/docs/tutorials/getting-started.md)` and `nix-develop` tries to rebuild GHC from scratch, but maybe it's on purpose.

> I'm having some difficulty seeing how that makes debugging easier or harder.
It basically makes a difference when you have other log messages:
```
[send] (set-option :produce-models true)
[warning] some log message
...
[info] some other log message
[send] (set-option :print-sucess true)) # intentional typo
[warning] some log message
...
[info] some other log message
[send] (assert (= 0 0))
[warning] some log message
...
[info] some other log message
[send] (check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
```
vs 
```
[warning] some log message
...
[info] some other log message
[warning] some log message
...
[info] some other log message
[warning] some log message
...
[info] some other log message
[send] (set-option :produce-models true)
(set-option :print-sucess true)) # intentional typo
(assert (= 0 0))
(check-sat)
[recv] (error "line 2 column 34: invalid command, '(' expected")
